src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceWideRecordReader.java

```diff
@@ -70,7 +70,7 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
   public SalesforceWideRecordReader initialize(
       InputSplit inputSplit, AuthenticatorCredentials credentials)
       throws IOException, InterruptedException {
-    List<Map<String, ?>> fetchedIdList = fetchBulkQueryIds(inputSplit, null);
+    List<Map<String, ?>> fetchedIdList = fetchBulkQueryIds(inputSplit, credentials);
```

Comment:
Move the super.initialize call outside fetchBulkQueryIds and call it before the fetchBulkQueryIds call here.

Reply:
Moved ```super.initialize```  before ```fetchBulkQueryIds``` and removed arguments from ```fetchBulkQueryIds``` as they are not required anymore in the method. [b5c9fe5d483b43f874cdf4ade9370479ca92b1c0]